skymp5-server/cpp/server_guest_lib/ActorsMap.cpp

```diff
@@ -0,0 +1,67 @@
+#include "ActorsMap.h"
+#include "Config.h"
+#include "MpActor.h"
+#include <sstream>
+#include <stdexcept>
+
+ActorsMap::ActorsMap()
+{
+  actorByUserId.resize(g_maxPlayers, nullptr);
+}
+
+void ActorsMap::Set(Networking::UserId userId, MpActor* actor)
+{
+  if (userId == Networking::InvalidUserId) {
+    throw std::runtime_error(
+      "Trying to insert Networking::InvalidUserId into ActorsMap");
+  }
+  if (!actor) {
+    throw std::runtime_error("Trying to insert nullptr into ActorsMap");
+  }
+  if (userId >= actorByUserId.size()) {
+    std::stringstream ss;
+    ss << "UserId " << userId << " is too big to be stored in ActorsMap";
+    throw std::runtime_error(ss.str());
+  }
+  Erase(userId);
+  Erase(actor);
```

Comment:
@nic11  

Not sure. I thought it is useful in case:

```c++
Actor *actor = ...
Actor *actor2 = ...
Networking::UserId id = 1;
ActorsMap map;

map.Set(id, actor);
map.Set(id, actor2);
```

Reply:
Added tests to confirm this line is useful.